gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergPartitionDatasetTest.java

```diff
@@ -0,0 +1,323 @@
+    List<DataFile> dataFiles = new ArrayList<>();
+    for (String srcFilePath : srcFilePaths) {
+      DataFile dataFile = Mockito.mock(DataFile.class);
+      Path dataFilePath = new Path(srcFilePath);
+      Path qualifiedPath = sourceFs.makeQualified(dataFilePath);
+      Mockito.when(dataFile.path()).thenReturn(dataFilePath.toString());
+      Mockito.when(sourceFs.getFileStatus(Mockito.eq(dataFilePath))).thenReturn(getFileStatus(qualifiedPath));
+      dataFiles.add(dataFile);
+    }
+    return dataFiles;
+  }
+
+  private static FileStatus getFileStatus(Path path) {
+    FileStatus fileStatus = new FileStatus();
+    fileStatus.setPath(path);
+    return fileStatus;
+  }
+
+  private static void verifyCopyEntities(Collection<CopyEntity> copyEntities, boolean sameSrcAndDestWriteLocation) {
+    String srcWriteLocationStart = SRC_FS_URI + SRC_WRITE_LOCATION;
+    String destWriteLocationStart = DEST_FS_URI + (sameSrcAndDestWriteLocation ? SRC_WRITE_LOCATION : DEST_WRITE_LOCATION);
+    String srcErrorMsg = String.format("Source Location should start with %s", srcWriteLocationStart);
+    String destErrorMsg = String.format("Destination Location should start with %s", destWriteLocationStart);
+    for (CopyEntity copyEntity : copyEntities) {
+      String json = copyEntity.toString();
+      if (isCopyableFile(json)) {
+        String originFilepath = getOriginFilePathAsStringFromJson(json);
+        String destFilepath = getDestinationFilePathAsStringFromJson(json);
+        Assert.assertTrue(originFilepath.startsWith(srcWriteLocationStart), srcErrorMsg);
+        Assert.assertTrue(destFilepath.startsWith(destWriteLocationStart), destErrorMsg);
+        String originFileName = originFilepath.substring(srcWriteLocationStart.length() + 1);
+        String destFileName = destFilepath.substring(destWriteLocationStart.length() + 1);
+        Assert.assertTrue(destFileName.endsWith(originFileName), "Incorrect file name in destination path");
+        Assert.assertTrue(destFileName.length() > originFileName.length() + 1,
+            "Destination file name should be longer than source file name as UUID is appended");
+      } else{
+        verifyPostPublishStep(json);
+      }
+    }
+  }
+
+  private static void verifyPostPublishStep(String json) {
+    String expectedCommitStep = "org.apache.gobblin.data.management.copy.iceberg.IcebergOverwritePartitionsStep";
+    String actualCommitStep = new Gson().fromJson(json, JsonObject.class)
+        .getAsJsonObject("object-data")
+        .getAsJsonObject("step")
+        .getAsJsonPrimitive("object-type")
+        .getAsString();
+    Assert.assertEquals(actualCommitStep, expectedCommitStep);
+  }
+
+  private static boolean isCopyableFile(String json) {
+    String objectType = new Gson().fromJson(json, JsonObject.class)
+        .getAsJsonPrimitive("object-type")
+        .getAsString();
+    return objectType.equals("org.apache.gobblin.data.management.copy.CopyableFile");
+  }
```

Comment:
make `IcebergDatasetTest::isCopyableFile` public.  if we need it a third time, we could ultimately move it to a testing utils class.

how about having `IcebergDatasetTest::verifyPostPublishStep` take the class name as a param?  then have one method each to verify the class name:
```
void verifyOverwritePartitionsStep(String json) {
  String expectedClassName = "org.apache.gobblin.data.management.copy.iceberg.IcebergOverwritePartitionsStep";
  verifyPostPublishStep(expectedClassName);
}
```